common/persistence/sql/sqlExecutionManager.go

```diff
@@ -183,102 +184,92 @@ func (m *sqlExecutionManager) GetWorkflowExecution(request *p.GetWorkflowExecuti
 		}
 	}
 
+	info, err := workflowExecutionInfoFromBlob(execution.Data, execution.DataEncoding)
+	if err != nil {
+		return nil, err
+	}
+
 	var state p.InternalWorkflowMutableState
 	state.ExecutionInfo = &p.InternalWorkflowExecutionInfo{
 		DomainID:                     execution.DomainID.String(),
 		WorkflowID:                   execution.WorkflowID,
 		RunID:                        execution.RunID.String(),
-		TaskList:                     execution.TaskList,
-		WorkflowTypeName:             execution.WorkflowTypeName,
-		WorkflowTimeout:              int32(execution.WorkflowTimeoutSeconds),
-		DecisionTimeoutValue:         int32(execution.DecisionTaskTimeoutMinutes),
-		State:                        int(execution.State),
-		CloseStatus:                  int(execution.CloseStatus),
-		LastFirstEventID:             execution.LastFirstEventID,
 		NextEventID:                  execution.NextEventID,
-		LastProcessedEvent:           execution.LastProcessedEvent,
-		StartTimestamp:               execution.StartTime,
-		LastUpdatedTimestamp:         execution.LastUpdatedTime,
-		CreateRequestID:              execution.CreateRequestID,
-		DecisionVersion:              execution.DecisionVersion,
-		DecisionScheduleID:           execution.DecisionScheduleID,
-		DecisionStartedID:            execution.DecisionStartedID,
-		DecisionRequestID:            execution.DecisionRequestID,
-		DecisionTimeout:              int32(execution.DecisionTimeout),
-		DecisionAttempt:              execution.DecisionAttempt,
-		DecisionTimestamp:            execution.DecisionTimestamp,
-		StickyTaskList:               execution.StickyTaskList,
-		StickyScheduleToStartTimeout: int32(execution.StickyScheduleToStartTimeout),
-		ClientLibraryVersion:         execution.ClientLibraryVersion,
-		ClientFeatureVersion:         execution.ClientFeatureVersion,
-		ClientImpl:                   execution.ClientImpl,
-		SignalCount:                  int32(execution.SignalCount),
-		HistorySize:                  execution.HistorySize,
-		CronSchedule:                 execution.CronSchedule,
+		TaskList:                     info.GetTaskList(),
+		WorkflowTypeName:             info.GetWorkflowTypeName(),
+		WorkflowTimeout:              info.GetWorkflowTimeoutSeconds(),
+		DecisionTimeoutValue:         info.GetDecisionTaskTimeoutMinutes(),
```

Comment:
I"m confused. Why is it in minutes?

Reply:
Good question. I don't know why - I must have read some code and thought DecisionTimeoutValue is in minutes. I hate when struct members represent numbers and have no units suffix - I will double confirm if DecisionTimeoutValue is actually seconds and if so, change the sqlblob also to be in seconds